packages/ra-core/src/test-ui/SimpleFormIterator.tsx

```diff
@@ -33,7 +33,10 @@ const DefaultAddItemButton = (
     return (
         <button
             type="button"
-            onClick={() => add()}
+            onClick={event => {
+                event.preventDefault();
+                add();
+            }}
```

Comment:
If this is necessary then you need to update the docs to mention it.

Also, I don't understand why this is necessary here but not in the ra-ui-materialui components. What's the difference?  :thinking:

Reply:
That's indeed not the correct fix. I just needed to add a `type="button"` to the clear button